File: 2225-find-players-with-zero-or-one-losses/2225-find-players-with-zero-or-one-losses.js
/**
 * @param {number[][]} matches
 * @return {number[][]}
 */
var findWinners = function(matches) {
    let arr = new Array(100001);
    for(let i=0;i<100001;i++){
        arr[i]=-1;
    }
    matches.forEach((e)=>{
        if(arr[e[0]]===-1) arr[e[0]]=0;
        if(arr[e[1]]===-1) arr[e[1]]=0;
        arr[e[1]]++;
    })
    
    let ans = new Array(2);
    for(let i=0;i<2;i++){
        ans[i] = new Array();
    }
    for(let i=1;i<100001;i++){
        if(arr[i]===0){
            ans[0].push(i);
        }
        if(arr[i]===1){
            ans[1].push(i);
        }
    }
    return ans;
};

File: 2225-find-players-with-zero-or-one-losses/2225-find-players-with-zero-or-one-losses.cpp
class Solution {
public:
    vector<vector<int>> findWinners(vector<vector<int>>& m) {
        map<int,int> g;
        for(int i=0;i<m.size();i++){
            g[m[i][1]]++;
        }
        vector<set<int>> ans(2);
        for(vector<int> t:m){
            int a = t[0],b = t[1];
            if(g[a]==0) ans[0].insert(a);
            else if(g[a]==1) ans[1].insert(a);
            if(g[b]==1) ans[1].insert(b);
        }
        vector<vector<int>> t1(2);
        int i=0;
        for(set<int> t:ans){
            for(int it:t) t1[i].push_back(it);
            i++;
        }
        return t1;
    }
};